Allows placing KVStore and update images on separate storage devices - to internal and external flash

VeijoPesonen opened the pull request:
Allows having KVStore in internal and update image in external flash

Fixes a bug where it has not been possible to have KVStore in internal
flash while an update image image has been kept in external storage.

[TDBStore] changes the default TDBStore location. Instead of placing the TDBStore after the application - plus two spare sectors if a new app image is bigger than the original - now it's kept at the end of the flash

### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@SeppoTakalo 
@yossi2le 
@teetak01 

### Release Notes

Fixes TDB_INTERNAL configuration - makes possible to have the KVStore inside internal flash while update images are kept in an external storage.
<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Allows having KVStore in internal and update image in external flash

Fixes a bug where it has not been possible to have KVStore in internal
flash while an update image image has been kept in external storage.

teetak01 reviewed (approved):
Looks good. You could update also the title.

Commit:
[TDBStore] changes the default TDBStore location

Thus far the default position has been after the application plus two
spare sectors. For simplicity and to have a predictable location for the
TDBStore with the default configuration the location is now switched to
the end of the flash. Two last sectors to be exact.
